MIGRATIONGUIDE.md

```diff
@@ -0,0 +1,363 @@
+change](https://stripe.com/docs/upgrades#2018-05-21), an `id` of a subscription in line item lost 
+its meaning as the subscription ID.  
+ 
+#### API requests 
+ 
+On the API requests, because this client library takes an untyped map as a parameter request to an 
+API method, eg. `Source#create(Map<String, Object> params)`, removed parameters due to API 
+upgrade will not be caught statically. For example, before [version change](https://stripe.com/docs/upgrades#2018-08-23)
+`Subscription#cancel` allows deferred cancellation with `at_period_end` request parameter, but 
+now removed in favor of `Subscription#update` with `cancel_at_period_end` parameter. Similarly, 
+there is extensive change in parameters for `PaymentIntent#create` and `PaymentIntent#confirm` 
+for beta users of `PaymentIntent` in [version change](https://stripe.com/docs/upgrades#2018-11-08)
+
+Parameter validation is another common change. For example, 
+ [version change](https://stripe.com/docs/upgrades#2018-10-31) requires `Customer#update` 
+ parameter `name` to be within 250 character-long. On the other hand, parameter validation can 
+ also become less restrictive, and the assumption you had on the valid returned response can be 
+ violated. For example, if you had relied on Stripe validation on uniqueness of SKU `attributes` during 
+ creation/update, and had built business logic around this assumption, after [version change](https://stripe.com/docs/upgrades#2018-09-06)
+  the validation is removed and SKUs returned can have duplicate attributes. Your code 
+  depending on SKU attribute uniqueness has to be updated. 
+  
+Parameters can also take new default values. Your old code may not explicitly pass the parameter 
+value, but after version upgrade when default value kicks in, your same API call now has a 
+different business implication. For example, in [version change](https://stripe.com/docs/upgrades#2018-05-21), 
+`Subscription#update` will take `trial_from_plan` as true if it is not explicitly specified.
+
+#### API errors
+
+Less obvious ones are changes related to JSON error object which is not directly exposed to you. 
+This library converts such failure into `StripeException` a subclass of Java `Exception`. For 
+example, in [version change](https://stripe.com/docs/upgrades#2017-12-14) `card_error` is 
+introduced, but that will be surfaced to you simply as a different message in `ApiException`. No 
+change on your part is required.
+
+Change in HTTP error codes, however, can affect your integration as a different type of 
+`StripeException` thrown. For example, in [version change](https://stripe.com/docs/upgrades#2016-10-19),
+ insufficient permission will return code 403 instead of 401. This means your code checking for 
+ `AuthenticationException` may require updates to `PermissionException` instead. Similarly, 429 is 
+ introduced after [version change](https://stripe.com/docs/upgrades#2015-09-08) instead of 400. 
```

Comment:
`after [version change](https://stripe.com/docs/upgrades#2015-09-08)` -> `after the [2015-09-08 version change](https://stripe.com/docs/upgrades#2015-09-08)`